### PR TITLE
feat: user command

### PR DESCRIPTION
**Pull request by WilsontheWolf**

### CODE REVIEW

File: src/commands/UserCommand.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+// set the class name, the export at the bottom and the file name to your desired command name (same as the one in the register function)
+class CommandName extends Command {
+	constructor() {
+		super("user"); // the name under which the bot internally stores your command (should be the same as the named set in `register`, must be unique)
+	}
+
+	async execute(interaction: CommandInteraction): Promise<void> {
+		// put the logic of your command here
+		// for example:
```
Comment (Chicken) [inline src/commands/UserCommand.ts:18]: remove the template comments k thx

File: src/commands/UserCommand.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+		const user = interaction.options.getUser("user");
+		if (!user) return interaction.reply("Please provide a user.");
```
Comment (Chicken) [inline src/commands/UserCommand.ts:20]: A required argument
```suggestion
		const user = interaction.options.getUser("user", true);
```

File: src/commands/UserCommand.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+		const member = interaction.guild?.members.cache.get(user.id);
```
Comment (Chicken) [inline src/commands/UserCommand.ts:21]: Fetch for uncached

File: src/commands/UserCommand.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+import { CommandInteraction, MessageEmbed } from "discord.js";
+import { SlashCommandBuilder } from "@discordjs/builders";
+
+const toUnix = (timestamp: number | Date) => {
+	if (timestamp instanceof Date) timestamp = timestamp.getTime();
+	return Math.floor(timestamp / 1000);
+};
+
+// set the class name, the export at the bottom and the file name to your desired command name (same as the one in the register function)
+class CommandName extends Command {
```
Comment (Chicken) [inline src/commands/UserCommand.ts:11]: read the comment

File: src/commands/UserCommand.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+				: "Not boosting :(";
+			if (member.joinedTimestamp)
+				embed.addField(
+					"Joined",
+					`<t:${toUnix(member.joinedTimestamp)}>\n(<t:${toUnix(member.joinedTimestamp)}:R>)`,
+					true
+				);
+			embed.addField("Boosting", boosting, true).addField("Roles", roles);
+		}
+		interaction.reply({ embeds: [embed] });
```
Comment (Chicken) [inline src/commands/UserCommand.ts:57]: Await to bubble up the error handling to the command handler

File: src/commands/UserCommand.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+		}
+		embed.addField("Id", user.id, true);
+		embed.addField("Created", `<t:${toUnix(user.createdTimestamp)}> (<t:${toUnix(user.createdTimestamp)}:R>)`);
+		embed.setTimestamp();
+		if (member) {
+			let roles = "‎";
+			member.roles.cache.forEach((role) => {
+				if (role.id === role.guild.id) return;
+				roles += `<@&${role.id}> `;
+			});
```
Comment (Chicken) [inline src/commands/UserCommand.ts:44]: This could probably be made nicer with map or reduce instead of forEach

**Review by Chicken — CHANGES_REQUESTED** [review]

File: src/commands/UserCommand.ts (hunk 6 above)
Comment (WilsontheWolf) [reply src/commands/UserCommand.ts:44]: I probbaly could but won't

**Review by Chicken — APPROVED** [review]

**Review by StrangeGirlMurph — APPROVED** [review]

**Pull request merged**